### DISCUSSION TITLE
Using the VPN.

### DISCUSSION THREAD

**Discussion by yodaluca23** [SideStore/SideStore#184] [General]
I use Cloudflare Warp VPN, but I would like to use the SideStore VPN constantly too, I've already made numerus product feedbacks to Apple about this a long time ago for something else, so is their a way Cloudflare could integrate this into their VPN app on IOS? or is their a way I'm not aware of two have 2 VPNs at the same time, or is their any solution to this? (Other then turning one off and the other on) Maybe the VPN could be turned into a DNS config? (Which does support multiple at once I believe) I have no idea....

Also why does it not work over data? Is their any progress being made on this?

**Comment by Spidy123222** [comment]
So currently 1.1.1.1 is a dns config on the vpn config in wireguard which works with internet through 1.1.1.1 and the vpn is through the other addresses which are all local addressing. You can change dns in your WiFi settings instead and just remove 1.1.1.1 which is cloudflare dns and use what you want in the vpn config or in WiFi settings. I have heard some use 2 vpns but I think that’s pretty hard to come by.

**Reply by yodaluca23** [reply]
Well, there is 1.1.1.1 the DNS and there is 1.1.1.1 Cloudflare Warp VPN, I'm talking about the Warp...

Sorry for the confusion I've updated the post...

**Comment by f1shy-dev** [comment]
![image](https://user-images.githubusercontent.com/56125930/209432652-81331451-34c8-406d-a31c-f2453fe88187.png)
Not sure how this works but something like this could work if we could find a wireguard supporting app that uses the personal VPN thing

**Comment by JJTech0130** [comment]
According to https://developer.apple.com/documentation/networkextension/personal_vpn,
> a VPN configuration that uses one of the built-in VPN protocols (IPsec or IKEv2)

so it wouldn't be possible to use that with WireGuard unfortunately.

**Comment by codexss** [comment]
This app can use multiple wireguard vpn at the same time, but I tried it and couldn't use Sideload refresh
https://apps.apple.com/app/id1442620678

**Comment by rational-stars** [comment]
Why don't I have Sidestore VPN

**Reply by yodaluca23** [reply]
You have to download the WireGuard app, and import the config file... and how does this have anything to do with this discussion???
